src/SAWScript/Crucible/LLVM/X86.hs

```diff
@@ -329,17 +330,61 @@ llvm_verify_x86 (Some (llvmModule :: LLVMModule x)) path nm globsyms checkSat se
                       , show $ W4.ppExpr off
                       ]
         noExtraValidityPred _ _ _ _ = return Nothing
+        defaultMacawExtensions_x86_64 = Macaw.macawExtensions
+          (Macaw.x86_64MacawEvalFn sfs) mvar
+          (mkGlobalMap . Map.singleton 0 $ preState ^. x86GlobalBase)
+          funcLookup
+          noExtraValidityPred
+        llvmPointerBlock = fst . C.LLVM.llvmPointerView
+        llvmPointerOffset = snd . C.LLVM.llvmPointerView
+        -- Compare pointers that are not valid LLVM pointers. Comparing the
+        -- offsets as unsigned bitvectors is not sound, because of overflow
+        -- (e.g. `base - 1` is less than `base`, but -1 is not less than 0 when
+        -- compared as unsigned). It is safe to allow a small negative offset,
+        -- because each pointer base is mapped to an address that is not in
+        -- the first page, which is never mapped on X86_64 Linux.
+        doPtrCmp ::
+          (sym -> W4.SymExpr sym (C.BaseBVType w) -> W4.SymExpr sym (C.BaseBVType w) -> IO (W4.SymExpr sym C.BaseBoolType)) ->
+          Macaw.PtrOp sym w (C.RegValue sym C.BoolType)
+        doPtrCmp f = Macaw.ptrOp $ \sym mem w xPtr xBits yPtr yBits x y -> do
```

Comment:
I suggest adding a bit more precision to the description of the semantics of this comparison.  As I understand it, this establishes a 1 (4k) page "buffer" around the allocation. It does this by adding 4096 to the offset of the pointer (thus, if some operation makes the pointer negative, it doesn't overflow to be max int according to the unsigned comparison). It actually adds this buffer to both pointers being compared to maintain the relation between them *assuming that neither pointer had already overflowed to a huge offset*.